File: frontend/src/types/marketing.ts
export type MarketingSectionId =
  | 'campaigns'
  | 'creatives'
  | 'content-planner'
  | 'posts-copy'
  | 'funnels'
  | 'whatsapp'
  | 'landing-pages'
  | 'templates'
  | 'analytics';

export interface MarketingProject {
  id: number;
  name: string;
  audience: string;
  objective: string;
  offer: string;
  tone: string;
  channels?: string | null;
  budget?: number | null;
  creative_project_id?: string | null;
  status?: string;
  published_at?: string | null;
  last_publish_error?: string | null;
  notes?: string | null;
  created_at: string;
}

export interface MarketingKpi {
  key: string;
  label: string;
  value?: number | null;
  variation?: number | null;
  status: string;
  message: string;
}

export interface MarketingOverview {
  status: string;
  projects_count: number;
  kpis: MarketingKpi[];
  message: string;
}

export interface MarketingDataStatus {
  status: string;
  message: string;
}

export interface MarketingAudience {
  id: number;
  name: string;
  city?: string | null;
  state?: string | null;
  country?: string | null;
  ticket?: string | null;
  source?: string | null;
  reservations?: string | null;
  behavior?: string | null;
  created_at: string;
}

export type MarketingAudienceCreate = Omit<MarketingAudience, 'id' | 'created_at'>;

export interface MarketingCalendarEvent {
  id: number;
  title: string;
  date: string;
  channel?: string | null;
  status: string;
  notes?: string | null;
  created_at: string;
}

export type MarketingCalendarEventCreate = Omit<MarketingCalendarEvent, 'id' | 'created_at'>;

export interface MarketingCampaignListItem {
  id: number;
  name: string;
  objective: string;
  audience: string;
  status: string;
  source: string;
  channels?: string | null;
  budget?: number | null;
  last_publish_error?: string | null;
}

export interface MarketingCampaignList {
  status: string;
  campaigns: MarketingCampaignListItem[];
  message: string;
}

export interface MarketingFunnel {
  status: string;
  stages: Array<{
    stage: string;
    count?: number | null;
    conversion_rate?: number | null;
    amount?: number | null;
  }>;
  message: string;
}

export interface MarketingProjectCreate {
  name: string;
  audience: string;
  objective: string;
  offer: string;
  tone?: string;
  channels?: string | null;
  budget?: number | null;
  creative_project_id?: string | null;
  status?: string;
  notes?: string;
}

export interface MarketingProjectUpdate {
  name?: string;
  audience?: string;
  objective?: string;
  offer?: string;
  tone?: string;
  channels?: string | null;
  budget?: number | null;
  creative_project_id?: string | null;
  status?: string;
  notes?: string;
}

export interface MarketingTemplateProfile {
  businessName: string;
  marketSegment: string;
  audience: string;
  offer: string;
  tone: string;
  platform: string;
}

export interface MarketingCampaignInput {
  business_name: string;
  market_segment: string;
  target_audience: string;
  campaign_goal: string;
  offer: string;
  tone: string;
  channel: string;
  campaign_type: string;
  budget_range: string;
  call_to_action: string;
}

export interface MarketingCampaignResult {
  campaign_headline: string;
  primary_copy: string;
  secondary_copy: string;
  cta_suggestions: string[];
  offer_angle: string;
  pain_points: string[];
  objections: string[];
  positioning_summary: string;
  creative_suggestion: string;
}

export interface CreativeIdeasInput {
  niche: string;
  audience: string;
  objective: string;
  platform: string;
  tone: string;
}

export interface CreativeIdea {
  title: string;
  concept: string;
  hook: string;
  cta: string;
}

export interface ContentPlanInput {
  business_type: string;
  target_audience: string;
  goal: string;
  frequency_per_week: string;
  content_pillars: string;
  timeframe: string;
  platform: string;
}

export interface ContentPlanResult {
  weekly_plan: string[];
  monthly_plan: string[];
  content_pillar_breakdown: string[];
  posting_suggestions: string[];
  campaign_dates: string[];
  grouped_ideas: Record<'attraction' | 'authority' | 'conversion' | 'retention', string[]>;
}

export interface PostCopyInput {
  content_type: string;
  audience: string;
  goal: string;
  product_service: string;
  tone: string;
  cta: string;
}

export interface PostCopyResult {
  main_copy: string;
  variations: string[];
  cta_lines: string[];
  hashtags: string[];
  hook_suggestion: string;
}

export interface FunnelInput {
  business: string;
  offer: string;
  target_audience: string;
  acquisition_channel: string;
  funnel_objective: string;
}

export interface FunnelStage {
  stage: string;
  content: string;
  messaging_goal: string;
  cta: string;
  objections: string;
  nurturing: string;
}

export interface FunnelResult {
  stages: FunnelStage[];
}

export interface WhatsAppFlowInput {
  business_type: string;
  customer_stage: string;
  objective: string;
  tone: string;
  offer: string;
}

export interface WhatsAppFlowResult {
  sequence: string[];
  follow_up_timing: string[];
  variations: string[];
  human_version: string;
  direct_version: string;
}

export interface LandingPageInput {
  business: string;
  audience: string;
  offer: string;
  promise: string;
  tone: string;
  cta_objective: string;
}

export interface LandingPageResult {
  hero_title: string;
  subtitle: string;
  benefit_bullets: string[];
  offer_section: string;
  objections_section: string[];
  faq: string[];
  final_cta: string;
  proof_section: string;
}

export interface MarketingTemplate {
  id: string;
  title: string;
  niche: string;
  description: string;
  included_assets: string[];
  profile: MarketingTemplateProfile;
}

export interface MarketingAnalytics {
  cards: Array<{ label: string; value: string; change: string }>;
  engagement_trend: Array<{ week: string; value: number }>;
  content_output_by_week: Array<{ week: string; posts: number }>;
  channel_usage: Array<{ channel: string; percentage: number }>;
  conversion_by_campaign_type: Array<{ type: string; percentage: number }>;
}

File: backend/app/schemas/marketing.py
from pydantic import BaseModel, ConfigDict, Field


class MarketingCampaignRequest(BaseModel):
    company_name: str
    audience: str
    objective: str
    offer: str
    tone: str = "premium"


class MarketingCampaignResponse(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    campaign: str
    copy_text: str = Field(serialization_alias="copy")
    cta: str
    ad_structure: str
    creative_suggestion: str


class MarketingTool(BaseModel):
    id: str
    name: str
    description: str
    active: bool = True


class MarketingProjectCreate(BaseModel):
    name: str
    audience: str
    objective: str
    offer: str
    tone: str = "premium"
    channels: str | None = None
    budget: float | None = None
    creative_project_id: str | None = None
    status: str = "draft"
    notes: str | None = None


class MarketingProjectRead(BaseModel):
    id: int
    name: str
    audience: str
    objective: str
    offer: str
    tone: str
    channels: str | None = None
    budget: float | None = None
    creative_project_id: str | None = None
    status: str = "draft"
    published_at: str | None = None
    last_publish_error: str | None = None
    notes: str | None = None
    created_at: str


class MarketingProjectUpdate(BaseModel):
    name: str | None = None
    audience: str | None = None
    objective: str | None = None
    offer: str | None = None
    tone: str | None = None
    channels: str | None = None
    budget: float | None = None
    creative_project_id: str | None = None
    status: str | None = None
    notes: str | None = None


class MarketingPublishRequest(BaseModel):
    channels: list[str] = Field(default_factory=list)


class MarketingPublishResponse(BaseModel):
    id: int
    status: str
    message: str
    required_providers: list[str] = Field(default_factory=list)
    missing_providers: list[str] = Field(default_factory=list)


class MarketingCopyRequest(BaseModel):
    prompt: str


class MarketingCopyResponse(BaseModel):
    model_config = ConfigDict(protected_namespaces=())

    copy_text: str = Field(serialization_alias="copy")


class MarketingImagePromptResponse(BaseModel):
    prompt: str


class MarketingContentRequest(BaseModel):
    project_id: int
    context: str
    type: str = "social_post"


class MarketingContentResponse(BaseModel):
    copies: list[str]
    cta: str
    hook: str
    hashtags: list[str] = Field(default_factory=list)


class MarketingDataStatus(BaseModel):
    status: str = "empty"
    message: str


class MarketingAudienceCreate(BaseModel):
    name: str
    city: str | None = None
    state: str | None = None
    country: str | None = None
    ticket: str | None = None
    source: str | None = None
    reservations: str | None = None
    behavior: str | None = None


class MarketingAudienceRead(MarketingAudienceCreate):
    id: int
    created_at: str


class MarketingCalendarEventCreate(BaseModel):
    title: str
    date: str
    channel: str | None = None
    status: str = "scheduled"
    notes: str | None = None


class MarketingCalendarEventRead(MarketingCalendarEventCreate):
    id: int
    created_at: str


class MarketingKpiRead(BaseModel):
    key: str
    label: str
    value: float | None = None
    variation: float | None = None
    status: str = "empty"
    message: str


class MarketingOverviewResponse(BaseModel):
    status: str
    projects_count: int
    kpis: list[MarketingKpiRead]
    message: str


class MarketingSeriesPoint(BaseModel):
    label: str
    receita: float | None = None
    investimento: float | None = None


class MarketingRevenueInvestmentResponse(BaseModel):
    status: str
    points: list[MarketingSeriesPoint] = Field(default_factory=list)
    message: str


class MarketingChannelRevenueRead(BaseModel):
    channel: str
    receita: float | None = None
    percentual: float | None = None
    roas: float | None = None
    leads: int | None = None
    reservas: int | None = None


class MarketingChannelRevenueResponse(BaseModel):
    status: str
    channels: list[MarketingChannelRevenueRead] = Field(default_factory=list)
    message: str


class MarketingFunnelStepRead(BaseModel):
    stage: str
    count: int | None = None
    conversion_rate: float | None = None
    amount: float | None = None


class MarketingFunnelResponse(BaseModel):
    status: str
    stages: list[MarketingFunnelStepRead] = Field(default_factory=list)
    message: str


class MarketingCampaignListItem(BaseModel):
    id: int
    name: str
    objective: str
    audience: str
    status: str
    source: str
    channels: str | None = None
    budget: float | None = None
    last_publish_error: str | None = None


class MarketingCampaignListResponse(BaseModel):
    status: str
    campaigns: list[MarketingCampaignListItem] = Field(default_factory=list)
    message: str
